pkg/apis/autoscaling/v1alpha1/pa_types.go

```diff
@@ -155,15 +164,56 @@ func (pa *PodAutoscaler) ScaleBounds() (min, max int32) {
 	return
 }
 
-func (pa *PodAutoscaler) MetricTarget() (target int32, ok bool) {
+// Target returns the target annotation value or false if not present.
+func (pa *PodAutoscaler) Target() (target int32, ok bool) {
 	if s, ok := pa.Annotations[autoscaling.TargetAnnotationKey]; ok {
 		if i, err := strconv.Atoi(s); err == nil {
+			if i < 1 {
+				return 0, false
+			}
 			return int32(i), true
 		}
 	}
 	return 0, false
 }
 
+// Window returns the window annotation value or false if not present.
+func (pa *PodAutoscaler) Window() (window time.Duration, ok bool) {
+	if s, ok := pa.Annotations[autoscaling.WindowAnnotationKey]; ok {
+		d, err := time.ParseDuration(s)
+		if err != nil {
+			return 0, false
+		}
+		if d < time.Second*6 {
+			return 0, false
+		}
+		return d, true
+	}
+	return 0, false
+}
+
+// WindowPanicPercentage returns panic window annotation value or false if not present.
+func (pa *PodAutoscaler) WindowPanicPercentage() (percentage float64, ok bool) {
+	percentage, ok = pa.annotationFloat64(autoscaling.WindowPanicPercentageAnnotationKey)
+	// Panic window cannot be bigger than stable window. Or vanishingly small
+	// (cannot divide by zero).
+	if !ok || percentage > 100.0 || percentage < 1.0 {
+		return 0, false
+	}
+	return percentage, ok
+}
+
+// TargetPanicPercentage return the panic target annotation value or false if not present.
+func (pa *PodAutoscaler) TargetPanicPercentage() (percentage float64, ok bool) {
+	percentage, ok = pa.annotationFloat64(autoscaling.TargetPanicPercentageAnnotationKey)
+	// Panic target cannot be the same size as the stable target. Otherwise
+	// we would alway be panicking.
+	if !ok || percentage < 110.0 {
```

Comment:
should this be `!ok || <= 100.0`? The 10% are valid room for configuration, no?

Reply:
Per comment above, the panic threshold should be *greater than* the stable threshold.  And if it's not greater than by enough, any variance in the metrics will cause a panic.  I'm validating it's at least 10% bigger, for what I consider sane values.

Reply:
NIT: Extract 110.0 and other condition(6 seconds) to constants?

Reply:
+1, just for the ability to better understand the meaning of the value.

Reply:
Done.